packages/kit/src/runtime/client/client.js

```diff
@@ -328,9 +304,8 @@ export function create_client({ target, base, trailing_slash }) {
 					scrollTo(0, 0);
 				}
 			}
-		} else {
-			// in this case we're simply invalidating
-			await tick();
+
+			if (!keepfocus) reset_focus();
```

Comment:
I don't fully grasp the implications of this, so I'll ask: what difference does it make to switch the order of execution?

Reply:
Focus management needs to happen after the DOM has been updated — previously, it would reset while DOM updates were still pending. We got away with it because we were focusing the `<body>` which is unchanged, but that won't do any more

Reply:
Actually I'm wrong — we _can_ focus immediately, it turns out. And we need to, because if we wait for `tick()` to resolve it will blow away any manual focus management. Gonna blame sleep deprivation again